build-one.sh

```diff
@@ -13,8 +13,18 @@ REPO="khronosgroup/docker-images"
 (
     cd $(dirname $0)
     DOCKERFILE=$1
-    shift
-    TAG=${DOCKERFILE#Dockerfile.}
-    TAG=${TAG%.dockerfile}
-    docker build "$@" . -f "$DOCKERFILE" -t "$REPO:$TAG"
+    DATE=$2
+    shift 2
+    if [ "$1" == "push" ]; then
+        OP=$1
+        shift
+    fi
+    docker build "$@" . -f "$DOCKERFILE.Dockerfile" \
+        --build-arg "VERSION=$DATE" \
+        -t "$REPO:$DOCKERFILE-latest" \
+        -t "$REPO:$DOCKERFILE-$DATE"
+    if [ "$OP" == "push" ]; then
+        docker push "$REPO:$DOCKERFILE-latest"
+        docker push "$REPO:$DOCKERFILE-$DATE"
+    fi
```

Comment:
Two `if`s like this are a bit ugly, let's see if we can think of something better.

Reply:
Should we perhaps always consume `$2`, and _force_ the caller to invoke eg:

```console
$ ./for-all.sh build
$ ./for-all.sh buildAndPush
```

Or something like that?